homeassistant/components/ambient_network/config_flow.py

```diff
@@ -0,0 +1,185 @@
+                user_input[CONFIG_LOCATION][CONFIG_LOCATION_RADIUS],
+                UnitOfLength.METERS,
+                UnitOfLength.MILES,
+            )
+            return await self.async_step_station()
+
+        schema: vol.Schema = self.add_suggested_values_to_schema(
+            vol.Schema(
+                {
+                    vol.Required(
+                        CONFIG_LOCATION,
+                    ): LocationSelector(LocationSelectorConfig(radius=True)),
+                }
+            ),
+            {
+                CONFIG_LOCATION: {
+                    CONFIG_LOCATION_LATITUDE: self.hass.config.latitude,
+                    CONFIG_LOCATION_LONGITUDE: self.hass.config.longitude,
+                    CONFIG_LOCATION_RADIUS: CONFIG_LOCATION_RADIUS_DEFAULT,
+                }
+            },
+        )
+
+        return self.async_show_form(
+            step_id=CONFIG_USER,
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_station(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the second step to select the station."""
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+
+            def parse_station(station: str) -> dict[str, str]:
+                mac_address, station_name = station.split(",")
+                return {
+                    ENTITY_STATION_NAME: station_name,
+                    ENTITY_MAC_ADDRESS: mac_address,
+                }
+
+            self._station = parse_station(user_input[CONFIG_STATION])
+            return await self.async_step_station_name()
+
+        client: OpenAPI = OpenAPI()
+        stations: list[dict[str, Any]] = await client.get_devices_by_location(
+            self._latitude, self._longitude, radius=self._radius
+        )
+
+        if len(stations) == 0:
+            return self.async_abort(reason="no_stations_found")
+
+        options: list[SelectOptionDict] = list[SelectOptionDict]()
+        for station in sorted(
+            stations, key=lambda s: s[API_STATION_INFO][API_STATION_NAME]
+        ):
+            station_name: str = station[API_STATION_INFO][API_STATION_NAME]
+            option: SelectOptionDict = SelectOptionDict(
+                label=f"{station_name}",
+                value=f"{station[API_STATION_MAC_ADDRESS]},{station_name}",
+            )
+            options.append(option)
+
+        schema: vol.Schema = vol.Schema(
+            {
+                vol.Required(CONFIG_STATION): SelectSelector(
+                    SelectSelectorConfig(
+                        options=options,
+                        multiple=False,
+                    ),
+                )
+            }
+        )
+
+        return self.async_show_form(
+            step_id=CONFIG_STATION,
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_station_name(
```

Comment:
So if I understand correctly we give the user an opportunity to set a name for the station? And we also get a station name from the API?

We only allow setting names in helpers. With some nice device setup and naming we can make sure it's easy for the user to rename stations to something they want. But we should name it something the user is able to recognise.

Reply:
Oh. I got it. How do we deal with the case where the name from the API is empty (which, unfortunately, I've seen happen)? Do we pick a default name? And is there an easy way to translate that default name (e.g., "Weather Station" -> "Wetter Station") in the config flow?

Reply:
There is no easy way to translate strings like that for the title. But users can easily rename them in the Ui (and with some other changes I think are needed for this integration) it would also rename the device directly to match the config entry

Reply:
Ok. Fixed.